### ISSUE TITLE
Actual alert text never returned.

### ISSUE THREAD

**Issue by afathonih** [appium/appium-espresso-driver#943]
Espresso server currently returning `UIObject2.toString()` instead of the actual text.

Log:
```
10-31 17:14:07.128 18386 18853 D appium  : Started processing GET request for '/session/e3fb49e9-e441-44a7-90a0-e5b9620dd80c/alert/text'
10-31 17:14:07.128 18386 18853 D appium  : Matched route definition: class io.appium.espressoserver.lib.http.RouteDefinition
10-31 17:14:07.128 18386 18853 D appium  : Matched route handler: class io.appium.espressoserver.lib.handlers.GetAlertText
10-31 17:14:07.128 18386 18853 I appium  : Executing GetAlertText handler
10-31 17:14:07.134 18386 18853 D appium  : Permission alert has been detected
10-31 17:14:07.137 18386 18853 D appium  : Detected 13 alert elements
10-31 17:14:07.154 18386 18853 D appium  : (added debug) alert text: androidx.test.uiautomator.UiObject2@38fdf
10-31 17:14:07.154 18386 18853 D appium  : androidx.test.uiautomator.UiObject2@39ee3
10-31 17:14:07.154 18386 18853 D appium  : androidx.test.uiautomator.UiObject2@3a2a4
10-31 17:14:07.154 18386 18853 D appium  : Finished processing GET request for '/session/e3fb49e9-e441-44a7-90a0-e5b9620dd80c/alert/text'
```

```
return alertElements.toTypedArray()
                    .filter { it.resourceName != null && !it.resourceName.matches(alertButtonsResIdPattern.toRegex()) }
                    .filter { (it.text ?: "").isNotEmpty() }
                    .joinToString(separator = "\n")
```

Fix:
```
return alertElements.toTypedArray()
                    .filter { it.resourceName != null && !it.resourceName.matches(alertButtonsResIdPattern.toRegex()) }
                    .filter { (it.text ?: "").isNotEmpty() }
                    .map { it.text } // bug fix
                    .joinToString(separator = "\n")
```

**Comment by mykola-mokhnach** [comment]
Thanks for the report. Should be done in https://github.com/appium/appium-espresso-driver/pull/944

**Comment by mykola-mokhnach** [comment]
the patch has been published in v2.28.1+
